dkg/pedersen/handler.go

```diff
@@ -530,3 +607,299 @@ func (h *Handler) announceDkgPublicKey(out mino.Sender, from mino.Address) error
 
 	return nil
 }
+
+// handles the resharing request - acts differently for the new and old and common nodes
+func (h *Handler) handleReshare(out mino.Sender,
+	from mino.Address, msg types.ResharingRequest, done chan struct{}) error {
+	// if the node is completely new
+	if h.startRes.distrKey == nil {
+		if h.startRes.Done() {
```

Comment:
this can never be the case. `Done()` actually verifies that `distrKey != nil`

Reply:
True. Sorry for the stupid mistake. I guess since both old and new nodes call `handleReshare` function I should completely delete these lines.